sgx-types/src/ssa.rs

```diff
@@ -273,12 +273,39 @@ pub struct Miscellaneous {
 /// The size of XSAVE region in SSA is derived from the enclave’s support of the collection
 /// of processor extended states that would be managed by XSAVE. The enablement of those
 /// processor extended state components in conjunction with CPUID leaf 0DH information
-/// determines the XSAVE region size in SSA.
+/// determines the XSAVE region size in SSA. See Section 38.9, Table 38-7.
 ///
-/// Section 38.9, Table 38-7
+/// See also Section 13.4 in
+/// https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-1-manual.pdf
+/// for details on the fields included in XSave.
 #[derive(Debug)]
 #[repr(C, align(4096))]
-pub struct XSave([[u64; 32]; 16]);
+pub struct XSave {
+    fcw: u16,
+    reserved0: [u32; 5],
+    mxcsr: u32,
+    mxcsr_mask: u32,
+    reserved1: [[u32; 32]; 3],
+    reserved2: [u32; 27],
+    xcomp_bv: u64,
+    reserved3: [u32; 12],
+}
```

Comment:
You appear to have filled in some of the XSave Legacy Region fields into the XSave struct. But:
1. You're missing many fields.
2. You've only covered the legacy region. You also need the XSAVE header and the extended area.
3. We're missing documentation (why isn't the docs error being thrown?).
4. The fields don't seem to correspond to the table 13-1.

Reply:
@npmccallum The fields are missing because I only filled in the minimal set from https://github.com/jsakkine-intel/linux-sgx/blob/master/tools/testing/selftests/x86/sgx/encl_bootstrap.S#L83, as suggested in #171. That's why it doesn't look like the table. 

However, I can add the rest of the fields from 13-1 now.